src/AbstractDifferentiation.jl

```diff
@@ -39,14 +61,37 @@ function derivative(ab::AbstractBackend, f, xs::Number...)
     end
 end
 
+"""
+    AD.gradient(ab::AD.AbstractBackend, f, xs...)
+
+Compute the gradients `gs` of `f` wrt the inputs `xs` using the backend `ab`.
+
+Return a tuple `gs` of gradients, one for each element in `xs`.
+"""
 function gradient(ab::AbstractBackend, f, xs...)
     return reshape.(adjoint.(jacobian(lowest(ab), f, xs...)),size.(xs))
 end
+
+"""
+    AD.jacobian(ab::AD.AbstractBackend, f, xs...)
+
+Compute the Jacobians `js` of `f` wrt the inputs `xs` using the backend `ab`.
+
+Return a tuple `js` of Jacobians, one for each element in `xs`.
+"""
 function jacobian(ab::AbstractBackend, f, xs...) end
+
 function jacobian(ab::HigherOrderBackend, f, xs...) 
     jacobian(lowest(ab), f, xs...)
 end
 
+"""
+    AD.hessian(ab::AD.AbstractBackend, f, x)
+
+Compute the Hessian `h` of `f` wrt the input `x` using the backend `ab`.
+
+Return a single matrix `h`, because `hessian` currently only supports a single input.
```

Comment:
```suggestion
Compute the Hessian of `f` wrt the input `x` using the backend `ab`.

The function returns a single matrix because `hessian` currently only supports a single input.
```